src/isotools/_transcriptome_stats.py

```diff
@@ -684,3 +687,103 @@ def rarefaction(self, groups=None, fractions=20, min_coverage=2, tr_filter={}):
+    :param transcriptome: the Transcriptome object on which the events were computed
+    :param gene: Gene object corresponding to the gene in which the event happened
+    :param min_dist: minimum distance (in nucleotides) between the two Alternative Splicing Events for the pair to be tested
+    :type min_dist: int
+    :param test: test to be performed. One of ("chi2", "fisher")
+    :type test: str
+    :param sg: segment graph
+    '''
+
+    coverage = gene.coverage.sum(axis=0) if len(gene.coverage.shape) > 1 else gene.coverage
+
+    C = pd.DataFrame({"A_pri": (0, 0), "A_alt": (0, 0)})
+
+    C.index = ("B_pri", "B_alt")
+
+    for m, n in itertools.product(range(2), range(2)):
+        tr_IDs = list(set(e1[m]) & set(e2[n]))
+        c_mn = sum([coverage[ID] for ID in tr_IDs])+0.01
+        C.iloc[n, m] = c_mn
+
+    TEST = TESTS[test]
+    test_res = TEST(C)
+
+    priA_priB, altA_altB = C.iloc[0, 0], C.iloc[1, 1]
+    priA_altB, altA_priB = C.iloc[1, 0], C.iloc[0, 1]
+    p_value = test_res[1]
+    test_stat = test_res[0]
+
+    return p_value, test_stat, int(priA_priB-0.01), int(priA_altB-0.01), int(altA_priB-0.01), int(altA_altB-0.01)
+
+
+def coordination_test(self, test="chi2", min_dist=1, min_total=100,
+                      min_alt_fraction=.1, event_type=("ES", "5AS", "3AS", "IR", "ME")):
+    '''
+    Performs gene_coordination_test on all genes.
+
+    :param test: Test to be performed. One of ("chi2", "fisher")
+    :type test: str
+    :param min_dist: Minimum distance (in nucleotides) between the two Alternative Splicing Events for the pair to be tested
+    :type min_dist: int
+    :param min_total: The minimum total number of reads for an event to pass the filter
+    :type min_total: int
+    :param min_alt_fraction: The minimum fraction of read supporting the alternative
+    :type min_alt_frction: float
+    :param event_type:  A tuple with event types to test. Valid types are (‘ES’,’3AS’, ‘5AS’,’IR’ or ‘ME’, ‘TSS’, ‘PAS’).
+    Default is ("ES", "5AS", "3AS", "IR", "ME")
+
+    :return: a Pandas dataframe, where each column corresponds to the p_values, the statistics
+    (the chi squared statistic if the chi squared test is used and the odds-ratio if the Fisher
+    test is used), the gene name, the type of the first ASE, the type of the second ASE, the
+    starting coordinate of the first ASE, the ending coordinate of the first ASE, the starting
+    coordinate of the second ASE, the ending coordinate of the second ASE,
+    and the four entries of the contingency table.
+    '''
+    p_value = []
+    stat = []
+    Gene = []
+    ASE1_type, ASE2_type = [], []
+    ASE1_start = []
+    ASE1_end = []
+    ASE2_start = []
+    ASE2_end = []
+    priA_priB = []
+    priA_altB = []
+    altA_priB = []
+    altA_altB = []
+
+    for g in self.gene_table().gene_name:
+        test_res = self[g].gene_coordination_test(test=test, min_dist=min_dist, min_total=min_total,
```

Comment:
gene_coordination_test should return a list of test result tuples, one tuple per comparison. This makes it much simpler
```suggestion
        test_res.extend(self[g].gene_coordination_test(test=test, min_dist=min_dist, min_total=min_total,
```

Reply:
Resolved. But then I still do 13 list comprehensions to later use in the construction of the pandas df. If frugality was your intention, I might have to look at different ways to construct a pandas df before merging. If better readability was what you were aiming for, I think it now still looks better than before.